lib/utils/layoutUtil.js

```diff
@@ -120,6 +120,15 @@ export function connectElements(source, target, layoutGrid) {
     }
   }
 
+  // negative dX indicates connection from future to past
+  if (dX < 0 && dY <= 0) {
+    return [
+      getDockingPoint(sourceMid, sourceBounds, 'b'),
+      { x: sourceMid.x, y: sourceMid.y + DEFAULT_CELL_HEIGHT / 2 },
+      { x: targetMid.x, y: sourceMid.y + DEFAULT_CELL_HEIGHT / 2 },
+      getDockingPoint(targetMid, targetBounds, 'b')
+    ];
+  }
```

Comment:
@nikku Not exactly sure if this is the right approach. I have added the condition when it's a loop (future to past) means dX will be negative and the connection can be either on the same row or row less than the current element (dY <=0). 

For reference, It solves this case.
<img width="1108" alt="image" src="https://github.com/user-attachments/assets/4ccca774-4e66-4304-ae66-a5bce7552884">